### PR TITLE
[4.0] [plg_system_httpheaders] Force http headers

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #25592 cc @brianteeman 

### Summary of Changes

rename the option to `Force HTTP Headers` and clearly state in the description what is happening here.

### Testing Instructions

Please note the new description of the option.

### Expected result

`Force HTTP Headers`

### Actual result

`Additional HTTP Headers`

### Documentation Changes Required

This is not documented yet but it should be ;)

### CODE REVIEW

File: administrator/language/en-GB/en-GB.plg_system_httpheaders.ini [hunk 1]
```diff
@@ -5,9 +5,9 @@
 
 ; Please do not translate the word 'HTTP Headers' in the following two language strings
 PLG_SYSTEM_HTTPHEADERS="System - HTTP Headers"
-PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER="Additional HTTP Headers"
+PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER="Force HTTP Headers"
 ; Please do not translate the name of the HTTP Headers in the following language string
-PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_DESC="The supported headers are: <br><ul><li>Strict-Transport-Security</li><li>Content-Security-Policy</li><li>Content-Security-Policy-Report-Only</li><li>X-Frame-Options</li><li>X-XSS-Protection</li><li>X-Content-Type-Options</li><li>Referrer-Policy</li><li>Expect-CT</li></ul>"
+PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_DESC="This option allows to force a specific header value. Please note that this option overwirtes any prevusly set headers. The supported headers are: <br><ul><li>Strict-Transport-Security</li><li>Content-Security-Policy</li><li>Content-Security-Policy-Report-Only</li><li>X-Frame-Options</li><li>X-XSS-Protection</li><li>X-Content-Type-Options</li><li>Referrer-Policy</li><li>Expect-CT</li></ul>"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_system_httpheaders.ini]:
```suggestion
PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_DESC="This option will force a specific header value and will replace any previously set headers. The supported headers are: <br><ul><li>Strict-Transport-Security</li><li>Content-Security-Policy</li><li>Content-Security-Policy-Report-Only</li><li>X-Frame-Options</li><li>X-XSS-Protection</li><li>X-Content-Type-Options</li><li>Referrer-Policy</li><li>Expect-CT</li></ul>"
```

**Comment by brianteeman** [comment]
It's still problematic
If you set "write headers to configuration file" and then options as in this screenshot
image

Then both referrer policy are written to the htaccess file 
image

Although only the second one will be used.

**Comment by brianteeman** [comment]
actually its a little more complicated than I wrote above. As sometimes you do only get the one line.

Its all very confusing. Writing headers to the file actually makes it more confusing (for me). Is it even compatible with setting headers for site or admin or both?

There is no way to delete all the headers from the htaccess. So once they have been written to the htaccess if you then disable that option no changes made to the settings will be applied

We also now have the situation that we have rules set in the htaccess, rules set in the plugin, and rules set in the component and no way to see in one place all the rules that are being applied. This will make debugging a faulty rule very hard.

**Comment by zero-24** [comment]
Can we please open a separate issue for the htaccess issue, so we can keep this here to the suggested rename of the feature?

**Comment by zero-24** [comment]
> Although only the second one will be used.

Isn't that what force implies?

**Comment by zero-24** [comment]
> Can we please open a separate issue for the htaccess issue, so we can keep this here to the suggested rename of the feature?

Done here: https://github.com/joomla/joomla-cms/issues/25716 so we can keep this on renaming the feature as suggested.

File: administrator/language/en-GB/en-GB.plg_system_httpheaders.ini [hunk 2]
```diff
@@ -5,9 +5,9 @@
 
 ; Please do not translate the word 'HTTP Headers' in the following two language strings
 PLG_SYSTEM_HTTPHEADERS="System - HTTP Headers"
-PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER="Additional HTTP Headers"
+PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER="Force HTTP Headers"
 ; Please do not translate the name of the HTTP Headers in the following language string
-PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_DESC="The supported headers are: <br><ul><li>Strict-Transport-Security</li><li>Content-Security-Policy</li><li>Content-Security-Policy-Report-Only</li><li>X-Frame-Options</li><li>X-XSS-Protection</li><li>X-Content-Type-Options</li><li>Referrer-Policy</li><li>Expect-CT</li></ul>"
+PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_DESC="This option will force a specific header value and will replace any previously set headers. The supported headers are: <br><ul><li>Strict-Transport-Security</li><li>Content-Security-Policy</li><li>Content-Security-Policy-Report-Only</li><li>X-Frame-Options</li><li>X-XSS-Protection</li><li>X-Content-Type-Options</li><li>Referrer-Policy</li><li>Expect-CT</li></ul>"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.plg_system_httpheaders.ini]: The label is self explanatory.; No need for the description.

Comment (zero-24) [reply administrator/language/en-GB/en-GB.plg_system_httpheaders.ini]: This desc is about to be removed here anyway https://github.com/joomla/joomla-cms/pull/25753 For that reason i'm reverting this here to the original version and we can merge the rename of the feature. When the #25753 gets merged the desc string gets removed too.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on dd28fac11348ccf1f92dc24f84cc8aa083916618<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25713">issues.joomla.org/tracker/joomla-cms/25713</a>.</sub>

**Comment by wilsonge** [comment]
Now that the files don't get edited this is fine. Merged

**Comment by wilsonge** [comment]
This makes sense and combined with the apache file no longer being edited I believe this solves the other part of the issue. Thanks!

**Comment by zero-24** [comment]
Doc change Done.

**Pull request merged**